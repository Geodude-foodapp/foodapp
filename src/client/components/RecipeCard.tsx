import React from 'react';

type RecipeCardProps = {
  recipeId: number;
  title: string;
  image: string;
  sourceurl: string;
  type: 'favorite' | 'search';
  addFavorite?: () => void;
  removeFavorite?: () => void;
};
export default ({
  recipeId,
  title,
  image,
  sourceurl,
  type,
  addFavorite,
  removeFavorite,
}: RecipeCardProps) => {
  const handleClick = type === 'favorite' ? removeFavorite : addFavorite;
  return (
    <div className='recipe-card'>
      <h1>{title}</h1>
      <img
        src={image}
        alt='recipe-image'
      />
      <a href={sourceurl}>Go to recipe site</a>
      <button onClick={handleClick}></button>
    </div>
  );
};
