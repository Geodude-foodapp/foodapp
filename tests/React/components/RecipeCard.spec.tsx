import { render } from '@testing-library/react';
import React from 'react';
import RecipeCard from '../../../src/client/components/RecipeCard';

describe('Recipe Card Component', () => {
  beforeAll(() => render(<RecipeCard />));
  it('matches the snapshot', () => {
    const recipeCard = document.querySelector('.recipe-card');
    expect(recipeCard).toMatchSnapshot();
  });
});
