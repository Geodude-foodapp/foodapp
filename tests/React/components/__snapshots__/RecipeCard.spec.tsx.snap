// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card Component matches the snapshot 1`] = `
<p
  class="recipe-card"
>
  Recipe Card
</p>
`;
